### PR TITLE
Remove the compatibility related server and client version checks in the code

### PR DESCRIPTION
**Pull request by ihsandemir**
Removed the server and client version dependent compatibility codes.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java [hunk 1]
```diff
@@ -951,13 +951,6 @@
     public static final HazelcastProperty DISCOVERY_SPI_PUBLIC_IP_ENABLED
             = new HazelcastProperty("hazelcast.discovery.public.ip.enabled", false);
 
-    /**
-     * When this property is true, if the server can not determine the connected client version, it shall assume that it is of
-     * 3.6.x version client. This property is especially needed if you are using ICache (or JCache).
-     */
-    public static final HazelcastProperty COMPATIBILITY_3_6_CLIENT_ENABLED
```
Comment (sancar) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java]: There was a counterpart of this on the client side. That should be deleted also.

**Comment by sancar** [comment]
Do we have any usages left of endpoint.getClientVersion() or conn.getConnectedServerVersion() ?

File: hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java (hunk 1 above)
Comment (ihsandemir) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java]: OK. Just removed and pushed.

**Comment by ihsandemir** [comment]
> Do we have any usages left of endpoint.getClientVersion() or conn.getConnectedServerVersion() ?

No usages for them, just removed and pushed.

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
run-lab-run

**Pull request merged**